### PR TITLE
separate quality presets for amd and nvenc

### PR DESCRIPTION
**Pull request by galister**
Includes nvenc dropdown for p1-p7
Amd dropdown for Speed, Quality, Balanced
Preset on top for Speed, Quality, Balanced

### CODE REVIEW

File: alvr/server/cpp/platform/linux/EncodePipeline.cpp [hunk 1]
```diff
@@ -14,6 +14,7 @@ extern "C" {
 
 void alvr::EncodePipeline::SetParams(FfiDynamicEncoderParams params) {
   if (params.updated) {
+    Info("SetParams: %u, %u @ %f", params.updated, params.bitrate_bps / 1'000'000UL, params.framerate);
```
Comment (zarik5) [inline alvr/server/cpp/platform/linux/EncodePipeline.cpp:17]: This should probably be removed.

File: alvr/session/src/settings.rs [hunk 2]
```diff
@@ -21,12 +21,24 @@ pub enum FrameSize {
...
+#[repr(u32)]
+#[derive(SettingsSchema, Serialize, Deserialize, Clone)]
+pub enum EncoderQualityPresetNvidia {
+    P1 = 0,
+    P2 = 1,
+    P3 = 2,
+    P4 = 3,
+    P5 = 4,
+    P6 = 5,
+    P7 = 6,
```
Comment (zarik5) [inline alvr/session/src/settings.rs:39]: I'm thinking it should be P1 = 1, P2 = 2 etc. Less surprising.

File: alvr/dashboard/src/dashboard/components/settings_controls/presets/builtin_schema.rs [hunk 3]
```diff
@@ -104,6 +104,39 @@ pub fn framerate_schema() -> PresetSchemaNode {
...
+            ("Speed", "Speed", "P1"),
+            ("Balanced", "Balanced", "P3"),
+            ("Quality", "Quality", "P5"),
+        ]
+        .into_iter()
+        .map(|(key, val_amd, val_nv)| HigherOrderChoiceOption {
+            display_name: key.into(),
+            modifiers: [
+                string_modifier(
+                    "session_settings.video.advanced_codec_options.nvenc_overrides.nvenc_quality_preset",
```
Comment (zarik5) [inline alvr/dashboard/src/dashboard/components/settings_controls/presets/builtin_schema.rs:122]: I haven't tested this, but I think you are missing a ".variant" at the end of the path. Same after amd_encoder_quality_preset below.

**Review by zarik5 — CHANGES_REQUESTED** [review]

**Comment by galister** [comment]
@zarik5 tested and ready

**Review by zarik5 — CHANGES_REQUESTED** [review]

**Comment by zarik5** [comment]
Alright that's good enough 👍

**Review by zarik5 — APPROVED** [review]

**Pull request merged**